lib/Runtime/Library/SubString.cpp

```diff
@@ -96,4 +96,12 @@ namespace Js
         return false;
     }
 
+    void SubString::CachePropertyRecord(_In_ PropertyRecord const* propertyRecord)
+    {
+        // Now that the property record owns a copy of the string data, transform
+        // this instance into a more efficient type.
+        this->originalFullStringReference = nullptr;
+        LiteralStringWithPropertyStringPtr* converted = LiteralStringWithPropertyStringPtr::ConvertString(this);
```

Comment:
Doesn't `SubString` not have enough space? It only has one additional pointer (the `originalFullStringReference`) but `LiteralStringWithPropertyStringPtr` has 2; the `propertyString` and the `propertyRecord`

Reply:
The allocator was giving us that extra space whether we wanted it to or not because of 16-byte alignment rules. I added an extra dummy pointer in SubString.

Reply:
Ah, I see, makes sense.